services/fluxmonitor_test.go

```diff
@@ -164,62 +168,233 @@ func TestNewFluxMonitor(t *testing.T) {
+			3 * time.Millisecond,
+			"50500",
+		},
+		{
+			"3 adapters, 1 non-expected response",
+			[]string{"50000", "51000", "wrong"},
+			0.01,
+			600,
+			15 * time.Millisecond,
+			3 * time.Millisecond,
+			"50500",
+		},
 	}
 	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			var mockAdapters []url.URL
-			for _, v := range tt.adapterResults {
-				mockAdapters = append(mockAdapters, *createMockAdapter(v))
-			}
+		var mockAdapters []url.URL
+		for _, v := range tt.adapterResults {
+			mockAdapters = append(mockAdapters, *createMockAdapter(v))
+		}
 
-			triggerJobRun := make(chan subscriber.Event, 10)
-			var fmConfig FluxMonitorConfig
+		triggerJobRun := make(chan subscriber.Event, 10)
+		var fmConfig FluxMonitorConfig
 
-			fmConfig.Adapters = mockAdapters
-			fmConfig.RequestData = `{"from":"BTC","to":"USD"}`
-			fmConfig.Multiply = 18
-			fmConfig.Threshold = tt.threshold
-			fmConfig.AbsoluteThreshold = tt.absoluteThreshold
-			fmConfig.Heartbeat = tt.heartbeat
-			fmConfig.PollInterval = tt.heartbeat
-			fmConfig.RuntimeConfig = store.RuntimeConfig{FMAdapterTimeout: 1 * time.Second, FMAdapterRetryAttempts: 1, FMAdapterRetryDelay: 1 * time.Second}
+		fmConfig.Adapters = mockAdapters
+		fmConfig.RequestData = `{"from":"BTC","to":"USD"}`
+		// fmConfig.Multiply = tt.multiply
+		fmConfig.Threshold = tt.threshold
+		fmConfig.AbsoluteThreshold = tt.absoluteThreshold
+		fmConfig.Heartbeat = tt.heartbeat
+		fmConfig.PollInterval = tt.heartbeat
+		fmConfig.RuntimeConfig = store.RuntimeConfig{FMAdapterTimeout: 100 * time.Millisecond, FMAdapterRetryAttempts: 1, FMAdapterRetryDelay: 10 * time.Millisecond}
 
+		t.Run("1 new round event tests: "+tt.name, func(t *testing.T) {
+			fmt.Printf("Testing %s", t.Name())
 			fm, err := NewFluxMonitor("test", fmConfig, triggerJobRun, &mockBlockchainManager{})
 			require.NoError(t, err)
 			defer fm.Stop()
 			wg := sync.WaitGroup{}
 			wg.Add(1)
 			fmt.Println(prettyPrint(fm.state))
-			fmt.Println("New round event, initiated: ", fm.state.RoundID+1)
+			fmt.Println("New round event, not initiated: ", fm.state.RoundID+1)
 			FAEvents <- common.FMEventNewRound{
 				RoundID:         fm.state.RoundID + 1,
 				OracleInitiated: false,
 			}
-
 			go func() {
 				defer wg.Done()
-				timeout := time.NewTimer(2 * time.Second)
-				defer timeout.Stop()
-
-				select {
-				case job := <-triggerJobRun:
-					fmt.Println("Job triggered", job["result"])
-					if tt.want == "no_job" {
-						t.Errorf("Job received. Want %s", tt.want)
-					}
-					if got := job["result"]; !reflect.DeepEqual(fmt.Sprintf("%s", got), fmt.Sprintf("%s", tt.want)) {
-						t.Errorf("GetTriggerJson() = %s, want %s", got, tt.want)
-					}
-
-				case <-timeout.C:
-					fmt.Println("Job timeout")
-					if tt.want != "no_job" {
-						t.Errorf("No Job received, timeout. Want %s", tt.want)
-					}
-				}
+				waitForTrigger(t, triggerJobRun, tt.want, COMMON_JOB_TRIGGER_TIMEOUT)
+			}()
+			wg.Wait()
```

Comment:
It looks like this block could be replaced with this:

```go
waitForTrigger(t, triggerJobRun, tt.want, COMMON_JOB_TRIGGER_TIMEOUT)
```

Or am I missing anything?

Reply:
It is replaced, isn't it?

Reply:
Ah, looks like the view here got a bit weird. I'll try again here:

```go
wg.Add(1)
go func() {
    defer wg.Done()
    waitForTrigger(t, triggerJobRun, tt.want, COMMON_JOB_TRIGGER_TIMEOUT)
}()
wg.Wait()
```

... would be the same as ....

```go
waitForTrigger(t, triggerJobRun, tt.want, COMMON_JOB_TRIGGER_TIMEOUT)
```

Reply:
Aha. Yes it is the same, nice!